### PR TITLE
Cleanup test suite

### PR DESCRIPTION
**Pull request by rgrinberg**
* Add more descriptions to opaque tests
* Remove a bunch of useless --display short

### CODE REVIEW

File: test/blackbox-tests/test-cases/dialects/run.t [hunk 1]
```diff
@@ -1,21 +1,17 @@
 Test the (dialect ...) stanza inside the dune-project file.
 
-  $ dune build --root good --display short @install
+  $ dune exec --root good ./main.exe
+  Entering directory 'good'
+  Entering directory 'good'
+
+  $ dune build --root good --display short @fmt
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/dialects/run.t:7]: Should this `--display short` be removed as well? The output contains some `.o` extensions. Another way to check that `fmt.exe` is being executed is to make `fmt.exe` print something.

File: test/blackbox-tests/test-cases/exes-with-c/run.t [hunk 2]
```diff
@@ -1,21 +1,9 @@
...
-      ocamlopt .aa.eobjs/native/dune__exe__Bb.{cmx,o}
-      ocamlopt bb.exe
-      ocamlopt aa.exe
-
-  $ _build/default/aa.exe
-  A
-
-  $ _build/default/bb.exe
-  B
+  $ dune build exec ./aa.exe
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/exes-with-c/run.t:1]:
```suggestion
  $ dune exec ./aa.exe
```

File: test/blackbox-tests/test-cases/exes-with-c/run.t [hunk 3]
```diff
@@ -1,21 +1,9 @@
...
+  Error: Don't know how to build exec
+  [1]
+  $ dune build exec ./bb.exe
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/exes-with-c/run.t:4]:
```suggestion
  $ dune exec ./bb.exe
```

File: test/blackbox-tests/test-cases/menhir/cmly/run.t [hunk 4]
```diff
@@ -1,14 +1,3 @@
...
-        ocamlc .test.eobjs/byte/test_menhir1.{cmi,cmti}
-      ocamlopt .test.eobjs/native/test_menhir1.{cmx,o}
-        ocamlc .test.eobjs/byte/lexer1.{cmi,cmo,cmt}
-        ocamlc .test.eobjs/byte/test.{cmi,cmo,cmt}
-      ocamlopt .test.eobjs/native/lexer1.{cmx,o}
-      ocamlopt .test.eobjs/native/test.{cmx,o}
-      ocamlopt test.exe
+  $ dune exec ./test.exe --debug-dependency-path
+  Fatal error: exception Failure("lexing: empty token")
+  [2]
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/menhir/cmly/run.t:3]: Not a huge deal that this test is failing given that we just want to test that `test.exe` is building, but it is a tad distracting. What about building and then using `ls` to check the exe is built? Or passing a valid input to the binary and make it print something.

File: test/blackbox-tests/test-cases/menhir/general-2.0/run.t [hunk 5]
```diff
@@ -1,29 +1,8 @@
...
-      ocamlopt src/.test.eobjs/native/test.{cmx,o}
-      ocamlopt src/test.exe
+Test the menhir extension version 2.0
+
+Note that the binary that we build fails when we run it. This doesn't matter to
+the test because we jsut care that we were able to build the binary.
+
+  $ dune exec ./src/test.exe --debug-dependency-path
+  Fatal error: exception Failure("lexing: empty token")
+  [2]
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/menhir/general-2.0/run.t:8]: Same as above. It's a bit clearer here with the comment though

File: test/blackbox-tests/test-cases/menhir/general/run.t [hunk 6]
```diff
@@ -1,23 +1,7 @@
...
-      ocamlopt src/.test.eobjs/native/lexer2.{cmx,o}
-      ocamlopt src/.test.eobjs/native/test.{cmx,o}
-      ocamlopt src/test.exe
+Build and run a source file that requires a menhir parser.
+
+(Running the executable fails, but it doesn't matter since we only care about compilation)
+
+  $ dune exec ./src/test.exe --debug-dependency-path
+  Fatal error: exception Failure("lexing: empty token")
+  [2]
```
Comment (jeremiedimino) [inline test/blackbox-tests/test-cases/menhir/general/run.t:7]: Here as well. I'd just do the `ls` check in all three cases, at least it's simple and we don't have to explain why it's ok for the test to fail.

**Comment by jeremiedimino** [comment]
BTW, should we enable tests for OSX in the CI? That would make tests that make too many assumptions about extensions fail early.

**Comment by rgrinberg** [comment]
@jeremiedimino I've altered the naming convention to what we discussed last week. A directory that contains a self contained test should now end with a `.t` extension.

**Pull request merged**